lib/puppet/type/scheduled_task.rb

```diff
@@ -92,32 +96,33 @@ def insync?(current)
       A trigger can contain the following keys:
 
       * For all triggers:
-          * `schedule` **(Required)** --- What kind of trigger this is.
-            Valid values are `daily`, `weekly`, `monthly`, or `once`. Each kind
-            of trigger is configured with a different set of keys; see the
-            sections below. (`once` triggers only need a start time/date.)
+          * `schedule` **(Required)** --- The schedule type. Valid values are
+            `daily`, `weekly`, `monthly`, or `once`.
           * `start_time` **(Required)** --- The time of day when the trigger should
             first become active. Several time formats will work, but we
             suggest 24-hour time formatted as HH:MM.
           * `start_date` ---  The date when the trigger should first become active.
             Defaults to the current date. You should format dates as YYYY-MM-DD,
             although other date formats may work. (Under the hood, this uses `Date.parse`.)
-      * For `daily` triggers:
```

Comment:
@Josh,

I tested with daily triggers, changing the clock to the future, everything kept working fine without re-applying. I do want to repeat a caveat from the old pull request (which is the same feature as the current provider has):
There is a small, minor issue though (which I don't care about too much but which should take into account). If you have a scheduled task with repeat triggers and for example a start time. Then you start managing this scheduled task with puppet, leave out the minutes_interval and minutes_duration definition it won't change the repeat triggers when nothing changes (which is what you want because by not defining those optional properties you leave them unmanaged), but when for example the start time changes, the repeat triggers will dissapear because the provider uses a dummy_trigger definition in which they initialize the minutes_interval and minutes_duration to 0. They should take the current values in the scheduled task instead if repeat triggers aren't managed. Actually the current provider has the same issue I guess.